packages/client/webapp/src/utils/petrinet-validator.ts

```diff
@@ -0,0 +1,121 @@
+import { IGraph, INode, IEdge } from 'graph-scaffolder';
+
+const extractEdgesAttribute = (
+	edges: IEdge[],
+	transitionOrState: string,
+	sourceOrTarget: string
+): string[] =>
+	edges
+		.filter((edge) => edge[sourceOrTarget].charAt(0) === transitionOrState)
+		.map((node) => node[sourceOrTarget]);
+
+/*
+    Validates petrinet - check #2 must come before check #5 to avoid an infinite loop
+*/
+export const petrinetValidator = (graph: IGraph, isBoundedPetrinet: boolean = true): boolean => {
+	const { nodes, edges }: { nodes: INode[]; edges: IEdge[] } = graph;
+
+	// console.log(graph);
+	// console.table(nodes);
+	// console.table(edges);
+
+	/* ----- 1. Requires at least one edge ----- */
+	if (edges.length < 1) {
+		console.log('#1');
```

Comment:
Maybe, instead of doing a `console.log()` you can return a message explaining why it is not valid.

Reply:
Agreed, wherever this validator is called will need to be able to access the reason why it failed to help gracefully handle the error.